tests/load/bigquery/test_bigquery_table_builder.py

```diff
@@ -371,3 +380,413 @@ def demo_source() -> DltResource:
+def drop_bigquery_schema() -> Iterator[None]:
+    yield
+    drop_active_pipeline_data()
+
+
+def test_adapter_no_hints_parsing() -> None:
+    @dlt.resource(columns=[{"name": "int_col", "data_type": "bigint"}])
+    def some_data() -> Iterator[Dict[str, str]]:
+        yield from next(sequence_generator())
+
+    assert some_data.columns == {
+        "int_col": {"name": "int_col", "data_type": "bigint"},
+    }
+
+
+def test_adapter_hints_parsing_partitioning_more_than_one_column() -> None:
+    @dlt.resource(
+        columns=[{"name": "col1", "data_type": "bigint"}, {"name": "col2", "data_type": "bigint"}]
+    )
+    def some_data() -> Iterator[Dict[str, Any]]:
+        yield from [{"col1": str(i), "col2": i} for i in range(3)]
+
+    assert some_data.columns == {
+        "col1": {"data_type": "bigint", "name": "col1"},
+        "col2": {"data_type": "bigint", "name": "col2"},
+    }
+
+    with pytest.raises(ValueError, match="^`partition` must be a single column name as a string.$"):
+        bigquery_adapter(some_data, partition=["col1", "col2"])
+
+
+def test_adapter_hints_parsing_partitioning() -> None:
+    @dlt.resource(columns=[{"name": "int_col", "data_type": "bigint"}])
+    def some_data() -> Iterator[Dict[str, str]]:
+        yield from next(sequence_generator())
+
+    bigquery_adapter(some_data, partition="int_col")
+    assert some_data.columns == {
+        "int_col": {"name": "int_col", "data_type": "bigint", "x-bigquery-column-partition": True},
+    }
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(all_staging_configs=True, subset=["bigquery"]),
+    ids=lambda x: x.name,
+)
+def test_adapter_hints_partitioning(destination_config: DestinationTestConfiguration) -> None:
+    @dlt.resource(columns=[{"name": "col1", "data_type": "bigint"}])
+    def no_hints() -> Iterator[Dict[str, int]]:
+        yield from [{"col1": i} for i in range(10)]
+
+    hints = bigquery_adapter(no_hints._clone(new_name="hints"), partition="col1")
+
+    @dlt.source(max_table_nesting=0)
+    def sources() -> List[DltResource]:
+        return [no_hints, hints]
+
+    pipeline = destination_config.setup_pipeline(
+        f"bigquery_{uniq_id()}",
+        full_refresh=True,
+    )
+
+    # noinspection PyArgumentList
+    pipeline.run(sources())
+
+    with pipeline.sql_client() as c:
+        with c.execute_query(
+            "SELECT ddl FROM `INFORMATION_SCHEMA.TABLES` WHERE table_name = 'hints'"
+        ) as cur:
+            hints_ddl: str = cur.fetchone()[0]
+            ast_hints = sqlglot.parse_one(hints_ddl, read="bigquery")
+            assert hints_ddl
+
+        with c.execute_query(
+            "SELECT ddl FROM `INFORMATION_SCHEMA.TABLES` WHERE table_name = 'no_hints'"
+        ) as cur:
+            no_hints_ddl: str = cur.fetchone()[0]
+            ast_no_hints = sqlglot.parse_one(no_hints_ddl, read="bigquery")
+            assert no_hints_ddl
+
+    diff = sqlglot.diff(ast_no_hints, ast_hints)
```

Comment:
It seems simpler and more reliable to use the bq client directly to inspect the generated schema like I did in other tests. Please advise

Reply:
you are free to kick that out! but let's say I'd ask you to add snowflake adapter. could you reuse part of this code? if not - you can kick it out

Reply:
The **ast** will look different from snowflakes, since they represent partitions differently. I don't see reuse value here.

I am kicking it out as advised

Reply:
Addressed in 1245d6d9d00d62b660af36e396fcb0524ad42c83